### DISCUSSION TITLE
Multiple Attachments with same Name

### DISCUSSION THREAD

**Discussion by bobskigit** [ravendb/ravendb#13773] [Q&A]
In raven studio UI if you have multiple attachments with the same name removing one, removes all.

I did not think you were allowed multiple attachments with the same name?

**Comment by ayende** [comment]
Can you give more details? You should not be able to have multiple attachments with the same name on a single document.

**Comment by bobskigit** [comment]
Sequence is 

 

Store attachment with same name as existing attachment

 

public async Task UpdateImage(string Id, IFormFile avatarImage,string overrideContentType = null)

        {

            await using var memoryStream = new MemoryStream();

            await avatarImage.CopyToAsync(memoryStream);

            memoryStream.Seek(0, SeekOrigin.Begin);

 

            // Get content type, not set if uploading from local file

            string contentType = !String.IsNullOrEmpty(overrideContentType) ? overrideContentType : avatarImage.ContentType;

            

            _dbSession.Advanced.Attachments.Store(channelId, Avatarkey, memoryStream, contentType);

 

 

If you then look in the studio at the attachments list for  this documentId you see two attachments with the same name. Press the remove icon and it removes both.

**Comment by haludi** [comment]
I tried to reproduce it with no success.
Can you submit a failing test? https://ravendb.net/docs/article-page/4.2/csharp/start/test-driver   
Can you send us export files before and after the duplication?
